modal_proto/api.proto

```diff
@@ -1341,6 +1344,8 @@ message S3Mount {
   string bucket_name = 1;
   string mount_path = 2;
   string credentials_secret_id = 3;
+  // if false, --allow-delete and --allow-overwrite are set
```

Comment:
I suggest removing this comment because it's an indirect reference to what we do internally.